core/tx_pool.go

```diff
@@ -1749,31 +1742,7 @@ func ValidateTransactorBalanceCoversTx(tx *types.Transaction, from common.Addres
 			return ErrInsufficientFunds
 		}
 	} else {
-		balance, err := currency.GetBalanceOf(currentVMRunner, from, *tx.FeeCurrency())
-		if err != nil {
-			log.Debug("ValidateTransactorBalanceCoversTx: error in getting fee currency balance", "feeCurrency", tx.FeeCurrency())
-			return err
-		}
-
-		if espresso {
-			// cost = GasFeeCap * gas + gatewayFee, as in (4)
-			cost := new(big.Int).SetUint64(tx.Gas())
-			cost.Mul(cost, tx.GasFeeCap())
-			if tx.GatewayFeeRecipient() != nil {
-				cost.Add(cost, tx.GatewayFee())
-			}
-			if balance.Cmp(cost) < 0 {
-				log.Debug("ValidateTransactorBalanceCoversTx: insufficient funds", "feeCurrency", tx.FeeCurrency(), "balance", balance)
-				return ErrInsufficientFunds
-			}
-		} else {
-			// cost = GasPrice * gas + gatewayFee, as in (3)
-			cost := tx.Fee()
-			if balance.Cmp(cost) <= 0 {
-				log.Debug("ValidateTransactorBalanceCoversTx: insufficient funds", "feeCurrency", tx.FeeCurrency(), "balance", balance)
-				return ErrInsufficientFunds
-			}
-		}
+		return erc20gas.TryDebitFees(tx, from, currentVMRunner)
```

Comment:
Rather than having an else statement we could simplify this function like this:

```go
if tx.FeeCurrency() == nil {
  return erc20gas.TryDebitFees(tx, from, currentVMRunner)
}
...
```